src/telemetry/mod.rs

```diff
@@ -62,27 +75,21 @@ pub mod metrics {
         );
     }
 
-    #[cfg(test)]
-    #[must_use]
-    pub fn get_counter_value<K: Into<KeyName>>(
-        snapshot: metrics_util::debugging::Snapshot,
-        metric_name: K,
-    ) -> Option<u64> {
-        use metrics_util::debugging::DebugValue;
-        use metrics_util::MetricKind;
-
-        let snapshot = snapshot.into_vec();
-        let metric_name = metric_name.into();
-
-        for (key, _unit, _, val) in snapshot {
-            if key.kind() == MetricKind::Counter && key.key().name() == metric_name.as_str() {
-                match val {
-                    DebugValue::Counter(v) => return Some(v),
-                    _ => unreachable!(),
-                }
-            }
-        }
+    // assumption: labels::STEP is set to "step" and labels::ROLE is set to "role"
+    // tracing macros take fields as literals, so I couldn't make it dynamic.
+    #[macro_export]
+    macro_rules! span {
```

Comment:
macro because I want to make sure `debug_span` is used consistently for metrics and variadic arguments (step + role, step or role) support makes it a bit nicer to write.